### PR TITLE
HPCC-13347 Add roxie workunit support to WsEcl

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by richardkchapman** [comment]
@wangkx please review

File: esp/xslt/wsecl3_form.xsl [hunk 1]
```diff
@@ -318,6 +326,12 @@ function switchInputForm()
 
                 <tr class='commands'>
                   <td align='left'>
+                  <xsl:if test="$includeRoxieOptions=1">
+                    <select id="job_type">
+                         <option value="QUERY">Call Query</option>
+                        <option value="WORKUNIT">Create Workunit</option>
```
Comment (wangkx) [inline esp/xslt/wsecl3_form.xsl]: Minor: indent

File: esp/xslt/wsecl3_xmltest.xsl [hunk 2]
```diff
@@ -361,7 +363,10 @@ function onSendRequest()
     document.getElementById("resp_body").value = "";
     document.getElementById("resp_header").value = "";
     
-    var url = "]]></xsl:text><xsl:value-of select="$destination"/><xsl:text disable-output-escaping="yes"><![CDATA[";
+   var url = "]]></xsl:text><xsl:value-of select="$destination"/><xsl:text disable-output-escaping="yes"><![CDATA[";
+   if (document.getElementById("job_type").value == "WORKUNIT")
+       url = "]]></xsl:text><xsl:value-of select="$createWorkunitDestination"/><xsl:text disable-output-escaping="yes"><![CDATA[";
```
Comment (wangkx) [inline esp/xslt/wsecl3_xmltest.xsl]: wrong indent

**Review by wangkx — COMMENTED** [review]
@afishbeck minor comment.

**Comment by afishbeck** [comment]
@wangkx updated

**Comment by wangkx** [comment]
@afishbeck looks good.

**Pull request merged**